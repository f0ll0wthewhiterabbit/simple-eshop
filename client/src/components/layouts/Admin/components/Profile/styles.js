import styled from 'styled-components'
import Avatar from '@material-ui/core/Avatar'
import teal from '@material-ui/core/colors/teal'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: ${props => props.theme.spacing(4)}px;
  padding-bottom: ${props => props.theme.spacing(4)}px;
`

export const ProfileAvatar = styled(Avatar)`
  width: ${props => props.theme.spacing(6)}px;
  height: ${props => props.theme.spacing(6)}px;
  margin-bottom: 5px;
  background-color: ${teal[500]};
  letter-spacing: -1px;
`
