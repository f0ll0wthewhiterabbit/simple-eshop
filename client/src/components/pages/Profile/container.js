import { connect } from 'react-redux'

import ProfilePage from './component'

const mapStateToProps = state => ({
  firstName: state.getIn(['auth', 'user', 'firstName']),
  lastName: state.getIn(['auth', 'user', 'lastName']),
})

export default connect(mapStateToProps)(ProfilePage)
