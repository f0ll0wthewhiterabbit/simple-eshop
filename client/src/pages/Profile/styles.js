import styled from 'styled-components'
import { Avatar, Typography } from '@material-ui/core'
import green from '@material-ui/core/colors/green'

export const Wrapper = styled.div`
  padding-top: ${props => props.theme.spacing(8)}px;
  padding-bottom: ${props => props.theme.spacing(12)}px;
  display: flex;
  flex-direction: column;
  align-items: center;
`

export const Heading = styled(Typography)`
  margin-bottom: ${props => props.theme.spacing(3)}px;
  font-size: 1.85rem;
  font-family: 'Montserrat', sans-serif;
  font-weight: 500;
`

export const IconWrapper = styled(Avatar)`
  margin: ${props => props.theme.spacing(1)}px;
  background-color: ${green[500]};
  width: ${props => props.theme.spacing(7)}px;
  height: ${props => props.theme.spacing(7)}px;
  font-size: 1.5rem;
  letter-spacing: -1px;
`
