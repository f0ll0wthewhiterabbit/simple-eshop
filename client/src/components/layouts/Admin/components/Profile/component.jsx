import React from 'react'
import PropTypes from 'prop-types'
import Typography from '@material-ui/core/Typography'

import { Wrapper, ProfileAvatar } from './styles'

const Profile = ({ firstName, lastName }) => {
  const firstLetter = firstName.charAt(0).toUpperCase()
  const secondLetter = lastName.charAt(0).toUpperCase()

  return (
    <Wrapper>
      <ProfileAvatar>{`${firstLetter} ${secondLetter}`}</ProfileAvatar>
      <Typography variant="h6">{`${firstName} ${lastName}`}</Typography>
      <Typography variant="caption">Administrator</Typography>
    </Wrapper>
  )
}

Profile.propTypes = {
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
}

export default Profile
