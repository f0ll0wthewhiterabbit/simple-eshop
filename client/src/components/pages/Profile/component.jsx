import React from 'react'
import PropTypes from 'prop-types'
import { Container } from '@material-ui/core'

import ProfileForm from './components/ProfileForm'
import { Wrapper, Heading, IconWrapper } from './styles'

const ProfilePage = ({ firstName, lastName }) => {
  const firstLetter = firstName.charAt(0).toUpperCase()
  const secondLetter = lastName.charAt(0).toUpperCase()

  return (
    <Container component="main" maxWidth="xs">
      <Wrapper>
        <Heading variant="h4" component="h1" align="center">
          Profile
        </Heading>
        <IconWrapper data-test="avatar">{`${firstLetter} ${secondLetter}`}</IconWrapper>
        <ProfileForm />
      </Wrapper>
    </Container>
  )
}

ProfilePage.propTypes = {
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
}

export default ProfilePage
