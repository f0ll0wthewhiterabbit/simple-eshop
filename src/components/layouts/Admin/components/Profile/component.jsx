import React from 'react'
import Typography from '@material-ui/core/Typography'
import PersonIcon from '@material-ui/icons/Person'

import { Wrapper, ProfileAvatar } from './styles'

const Profile = () => {
  const user = {
    name: 'John Doe',
    role: 'Administrator',
    avatar: <PersonIcon />,
  }

  return (
    <Wrapper>
      <ProfileAvatar>{user.avatar}</ProfileAvatar>
      <Typography variant="h6">{user.name}</Typography>
      <Typography variant="caption">{user.role}</Typography>
    </Wrapper>
  )
}

export default Profile
