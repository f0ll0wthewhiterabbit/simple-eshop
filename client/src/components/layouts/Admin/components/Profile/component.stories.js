import React from 'react'
import Profile from './component'

export default {
  title: 'Profile',
  component: Profile,
  excludeStories: /.*Data$/,
}

export const profileData = {
  firstName: 'John',
  lastName: 'Doe',
}

export const normal = () => <Profile {...profileData} />
