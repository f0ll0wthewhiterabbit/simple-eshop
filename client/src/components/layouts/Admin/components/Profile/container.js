import { connect } from 'react-redux'

import Profile from './component'

const mapStateToProps = state => ({
  firstName: state.getIn(['auth', 'user', 'firstName']),
  lastName: state.getIn(['auth', 'user', 'lastName']),
})

export default connect(mapStateToProps)(Profile)
