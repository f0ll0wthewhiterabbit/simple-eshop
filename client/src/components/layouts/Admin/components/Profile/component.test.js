import React from 'react'
import { shallow } from 'enzyme'

import Profile from './component'
import findByTestAttr from '../../../../../utils/findByTestAttr'

describe('Profile component', () => {
  let wrapper
  const initialProps = {
    firstName: 'John',
    lastName: 'Doe',
  }
  const dataTestAvatar = 'avatar'
  const dataTestName = 'name'

  const generateWrapper = passedProps => {
    const defaultProps = { ...initialProps }
    const props = { ...defaultProps, ...passedProps }

    return shallow(<Profile {...props} />)
  }

  beforeEach(() => {
    wrapper = generateWrapper()
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should contain correct abbreviation in avatar', () => {
    const avatar = findByTestAttr(wrapper, dataTestAvatar)

    expect(avatar.text()).toBe('J D')
  })

  it('should contain correct name and surname', () => {
    const name = findByTestAttr(wrapper, dataTestName)

    expect(name.text()).toBe('John Doe')
  })
})
