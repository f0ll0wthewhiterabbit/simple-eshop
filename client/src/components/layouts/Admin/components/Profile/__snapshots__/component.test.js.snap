// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component should render correctly 1`] = `
<styled.div>
  <Styled(WithStyles(ForwardRef(Avatar)))
    data-test="avatar"
  >
    J D
  </Styled(WithStyles(ForwardRef(Avatar)))>
  <WithStyles(ForwardRef(Typography))
    data-test="name"
    variant="h6"
  >
    John Doe
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    variant="caption"
  >
    Administrator
  </WithStyles(ForwardRef(Typography))>
</styled.div>
`;
